authz/providers/azure/rbac/checkaccessreqhelper.go

```diff
@@ -0,0 +1,234 @@
+func getUserId(userName string) string {
+	switch userName {
+	case "krdhruva@microsoft.com":
+		return "63e8a863-9ae9-4f3c-b0b7-fd9df05c712e"
+	case "test@KDOrg.onmicrosoft.com":
+		return "62103f2e-051d-48cc-af47-b1ff3deec630"
+	default:
+		return "62103f2e-051d-48cc-af47-b1ff3deec630"
+	}
+}
+
+func getScope(resourceId string, attr *authzv1.ResourceAttributes) string {
+	if attr != nil && attr.Namespace != "" {
+		return resourceId + "/namespace/" + attr.Namespace
+	}
+	return resourceId
+}
+
+func IsValidUUID(u string) bool {
+	_, err := uuid.Parse(u)
+	return err == nil
+}
+
+func getSecGroups(groups []string) []string {
+	var finalGroups []string
+	for _, element := range groups {
+		if IsValidUUID(element) {
+			finalGroups = append(finalGroups, element)
+		}
+	}
+	return finalGroups
+}
+
+func getActionName(verb string) string {
+	switch verb {
+	case "get":
+		fallthrough
+	case "list":
+		fallthrough
+	case "watch":
+		return "read"
+	case "create":
+		return "action"
+	case "update":
+		return "write"
+	case "delete":
+		return "delete"
+	default:
+		return ""
+	}
+}
+
+func getDataAction(subRevReq *authzv1.SubjectAccessReviewSpec, clusterType string) AuthorizationActionInfo {
+	authInfo := AuthorizationActionInfo{
+		IsDataAction: true}
+
+	authInfo.AuthorizationEntity.Id = clusterType
+	if subRevReq.ResourceAttributes != nil {
+		if subRevReq.ResourceAttributes.Group != "" {
+			authInfo.AuthorizationEntity.Id += "/" + subRevReq.ResourceAttributes.Group
+		}
+		authInfo.AuthorizationEntity.Id += "/" + subRevReq.ResourceAttributes.Resource + "/" + getActionName(subRevReq.ResourceAttributes.Verb)
+	} else if subRevReq.NonResourceAttributes != nil {
+		authInfo.AuthorizationEntity.Id += subRevReq.NonResourceAttributes.Path + "/" + getActionName(subRevReq.NonResourceAttributes.Verb)
+	}
+	return authInfo
+}
+
+func PrepareCheckAccessRequest(req *authzv1.SubjectAccessReviewSpec, clusterType, resourceId string) *CheckAccessRequest {
+	checkaccessreq := CheckAccessRequest{}
+	checkaccessreq.Subject.Attributes.ObjectId = getUserId(req.User)
+
+	if req.Groups != nil && len(req.Groups) > 0 {
+		groups := getSecGroups(req.Groups)
+		if groups != nil && len(groups) > 0 {
+			checkaccessreq.Subject.Attributes.Groups = groups
+			checkaccessreq.Subject.Attributes.ExpandGroupMembership = true
+		}
+	}
+
+	tmp := make([]AuthorizationActionInfo, 1)
+	tmp[0] = getDataAction(req, clusterType)
+	checkaccessreq.Actions = tmp
+	checkaccessreq.Resource.Id = getScope(resourceId, req.ResourceAttributes)
+
+	return &checkaccessreq
+}
+
+func getNameSpaceScope(req *authzv1.SubjectAccessReviewSpec, str *string) bool {
+	if req.ResourceAttributes != nil && req.ResourceAttributes.Namespace != "" {
+		*str = "/namespace/" + req.ResourceAttributes.Namespace
+		return true
+	}
+	return false
+}
+
+func ConvertCheckAccessResponse(body []byte) (*authzv1.SubjectAccessReviewStatus, error) {
+	var response []AuthorizationDecesion
```

Comment:
prefer
```golang
var (
...
)
```

Reply:
> []AuthorizationDecesion
just curious, why is it an array?

Reply:
I think array is not required.

Reply:
checkaccess supports multiple action check in one request and it send array as result.